lib/shared-sku-app/src/Editor/Field.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import { Button, Note, TextLink } from '@contentful/forma-36-react-components';
+import tokens from '@contentful/forma-36-tokens';
+import get from 'lodash/get';
+import isNil from 'lodash/isNil';
+import { css } from 'emotion';
+import { FieldExtensionSDK } from 'contentful-ui-extensions-sdk';
+import { SortableComponent } from './SortableComponent';
+import { ProductPreviewFn, OpenDialogFn, DisabledPredicateFn } from '../interfaces';
+
+interface Props {
+  sdk: FieldExtensionSDK;
+  cta: string;
+  logo: string;
+  fetchProductPreview: ProductPreviewFn;
+  openDialog: OpenDialogFn;
+  isDisabled: DisabledPredicateFn;
+}
+
+interface State {
+  value: string[] | string;
+  valid: boolean;
+  editingDisabled: boolean;
+}
+
+const styles = {
+  sortable: css({
+    marginBottom: tokens.spacingM
+  }),
+  container: css({
+    display: 'flex'
+  }),
+  logo: css({
+    display: 'block',
+    width: '30px',
+    height: '30px',
+    marginRight: tokens.spacingM
+  })
+};
+
+function getEmptyValue(sdk: FieldExtensionSDK) {
+  const isArray = sdk.field.type === 'Array';
+  return isArray ? [] : '';
+}
+
+export default class Field extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const value = props.sdk.field.getValue();
+    const isValidFieldType =
+      typeof value === 'string' ||
+      (Array.isArray(value) && value.every(v => typeof v === 'string'));
+
+    // `valid` is `true` if the app can render/write the value safely.
+    // If for example there is an object stored we don't want to override
+    // it without a user explicitly telling us to do so.
+    const valid = isNil(value) || isValidFieldType;
```

Comment:
I don't think this check is needed. We needed it in DAM apps because they use free-form Object fields. In this one we only deal with strings. `isValidFieldType` is not needed for sure because `fieldTypes` of `AppDefinition` will only include field types on which this app can operate.

Reply:
Removed.